lib/puma/thread_pool.rb

```diff
@@ -85,6 +85,17 @@ def self.clean_thread_locals
       end
     end
 
+    # generate stats hash so as not to perform multiple locks
+    # @return [Hash] hash containing stat info from ThreadPool
+    def stats
+      with_mutex do
+        { backlog: @todo.size,
+          running: @spawned,
+          pool_capacity: @waiting + (@max - @spawned)
+        }
+      end
```

Comment:
here's the approach i was thinking, in order to avoid repeating code (naming can change of course). WDYT?

```ruby
def backlog_logic
  @todo.size
end
def backlog
  with_mutext { backlog_logic }
end
def stats
  with_mutex do
    { backlog: backlog_logic,
      ...
    }
  end
end
```

Reply:
Actually, the three methods exist only so `Server#stats` can obtain the information.  There is one exception, and that may be removed with work currently being looked at to help with the 'long tail' issue.

So, thanks for the example.  As I mentioned, they may be deprecated and removed in Puma 7.

Reply:
ahh gotcha, i thought others were used for business logic like busy_threads